extensions/amp-inline-gallery/1.0/pagination.js

```diff
@@ -15,42 +15,61 @@
  */
 
 import * as Preact from '../../../src/preact';
-import * as styles from './pagination.css';
 import {CarouselContext} from '../../amp-base-carousel/1.0/carousel-context';
 import {useContext} from '../../../src/preact';
+import {useStyles} from './pagination.jss';
 
 /**
  * @param {!InlineGalleryDef.PaginationProps} props
  * @return {PreactDef.Renderable}
  */
 export function Pagination({inset, ...rest}) {
+  const classes = useStyles();
   const {slideCount, currentSlide, setCurrentSlide} = useContext(
     CarouselContext
   );
   const Comp = slideCount <= 8 ? Dots : Numbers;
   const indicator = (
     <div
       aria-hidden="true"
-      class="i-amphtml-carousel-pagination-container"
+      class={`${classes.container} ${inset ? classes.inset : ''}`}
```

Comment:
Do you prefer adding a class for `inset` as opposed to targeting the selector based on the `inset` attribute?

Reply:
Is `classes.inset` needed at all? It is `{}` in the jss file

Reply:
It's a marker on top-level and used throughout as a compound selector. E.g.

```
const container = {
  ...,
  '&$inset': {
    position: 'absolute',
    ...
  }
}
```

Which translates to:

```
.container {
  ...
}
.container.insert {
  position: absolute;
}
```

I wanted to try this way because:
1. It's close to the way we define these styles in 0.1. I used `.x.inset` instead of `.inset .x` here, mostly because it's faster in runtime.
2. A simple CSS specificity bump is less of a concern for inset styles because inset is already a specialized flag.
3. Byte-to-byte it's a more efficient since fewer styles need to be duplicated vs a `{inset ? nonInsetClass : insertClass}` approach.